include/mbedtls/config.h

```diff
@@ -1701,6 +1740,37 @@
  */
 //#define MBEDTLS_SSL_TRUNCATED_HMAC_COMPAT
 
+/**
+ * Allow SHA-1 in the default TLS configuration for certificate signing.
+ * Without this build-time option, SHA-1 support must be activated explicitly
+ * through mbedtls_ssl_conf_cert_profile. Turning on this option is not
+ * recommended because of it is possible to generate SHA-1 collisions, however
+ * this may be safe for legacy infrastructure where additional controls apply.
+ *
+ * \warning   SHA-1 is considered a weak message digest and its use constitutes
+ *            a security risk. If possible, we recommend avoiding dependencies
+ *            on it, and considering stronger message digests instead.
+ *
+ */
+// #define MBEDTLS_TLS_DEFAULT_ALLOW_SHA1_IN_CERTIFICATES
```

Comment:
This commit ("Move config.h boolean options to their appropriate section") is doing more than just moving options around. It has also explicitly disabled allowing SHA-1 in certs. I think that's a big enough change that it deserves its own commit and a changelog entry.

Reply:
If it had done that, that would be wrong. I do not intend to change any default setting in this PR. But I don't see the problem. This commit only reorders lines.
```
diff <(git show c00d4a138b6543cfceee402a175aae21d2e92ceb~1:include/mbedtls/config.h |sort) <(git show c00d4a138b6543cfceee402a175aae21d2e92ceb:include/mbedtls/config.h |sort)
```

Reply:
I misread as "MBEDTLS_TLS_DEFAULT_ALLOW_SHA1_IN_KEY_EXCHANGE", sorry.